astropy/cosmology/core.py

```diff
@@ -345,12 +345,16 @@ def __eq__(self, other: Any, /) -> bool:
         if other.__class__ is not self.__class__:
             return NotImplemented  # allows other.__eq__
 
-        # check all parameters in 'other' match those in 'self'
-        equivalent = self.__equiv__(other)
+        # check all parameters in 'other' match those in 'self' and 'other' has
+        # no extra parameters (latter part should never happen b/c same class)
+        # TODO! element-wise when there are array cosmologies
+        params_eq = (set(self.__all_parameters__) == set(other.__all_parameters__)
+                     and all(np.all(getattr(self, k) == getattr(other, k))
+                             for k in self.__all_parameters__))
```

Comment:
Is the following code significantly faster?
```suggestion
        try:
            params_eq = (len(self.__all_parameters__) == len(other.__all_parameters__)
                         and all(k == k and np.all(getattr(self, k) == getattr(other, k))
                                 for k in self.__all_parameters__))
         except KeyError:
             return False
```

Reply:
In my limited testing it's a wash. The issue is the ``k==k`` check doesn't do anything. With this construction ``other. __all_parameters__`` can be the same length but not have the same parameters, so long as they are define on ``other``.